### PR TITLE
Add fixes for changelog pull for zone records vs zones

### PR DESCRIPTION
**Pull request by abeeson**
Changes proposed in this pull request:

- Remove zones from changelog SQL call
- Improve handling for non change log entries
- Add extra protection from accidental zone insertions causing mismatch against zone_records

### CODE REVIEW

File: server/lib/NicToolServer/Export/BIND/nsupdate.pm [hunk 1]
```diff
@@ -488,9 +488,9 @@ sub zr_uri {
 
 sub zr_caa {
     my ( $self, $r, $mode ) = @_;
-    $mode = "add" unless defined $mode;
+    $mode = "add" unless defined($mode);
 
-    $r->{zone} = $self->{nte}->{zone_name} unless defined $r->{zone};
+    $r->{zone} = $self->{nte}->{zone_name} unless defined( $r->{zone} );
```
Comment (msimerson) [inline server/lib/NicToolServer/Export/BIND/nsupdate.pm:493]: I'm curious, why add parens here?

**Comment by msimerson** [comment]
If I'm reading this correctly, you are expanding the criteria for when an update is detected. Previously only zone_record updates where considered. Correct?

File: server/lib/NicToolServer/Export/BIND/nsupdate.pm (hunk 1 above)
Comment (abeeson) [reply server/lib/NicToolServer/Export/BIND/nsupdate.pm:493]: I was actually wondering the same. When I went through my original didn't have it but on merge it was complaining about that being different, so I matched it to the other entries that had it. Happy to remove them if you'd like

**Comment by abeeson** [comment]
Other way around, this is to prevent zone matches coming through, which were originally erroneously included in the select from the log.

We hit a problem where we've had some zone changes come through that have resulted in lookups against very old zone records, which obviously shouldn't occur.

File: server/lib/NicToolServer/Export/BIND/nsupdate.pm (hunk 1 above)
Comment (msimerson) [reply server/lib/NicToolServer/Export/BIND/nsupdate.pm:493]: That's funny, b/c in the next change up, you removed a pair. Removing them is preferred as they're no longer necessary in modern perl.

Comment (abeeson) [reply server/lib/NicToolServer/Export/BIND/nsupdate.pm:493]: no problem, I'll adjust them and push up the change.

Comment (abeeson) [reply server/lib/NicToolServer/Export/BIND/nsupdate.pm]: done, fixed up those on the lines above too, as well as all the older entries above. They should be consistent now

**Comment by msimerson** [comment]
Oh right, I read the diff wrong.  You're removing zones from the list of global log entries returned. But then, if a zone property changes (like the serial, TTL, or whatnot, then that change isn't going to be detected. Not a problem?

**Comment by abeeson** [comment]
Not a problem for this, the zones are defined and managed separately, and nsupdate is just inserting dynamic changes. The whole nsupdate mechanism falls apart if you are trying to manage the zones that way whilst also doing the zone records.

**Review by msimerson — APPROVED** [review]
LGTM

**Pull request merged**